### DISCUSSION TITLE
Developer feedback wanted!

### DISCUSSION THREAD

**Discussion by nagyesta** [nagyesta/lowkey-vault#272] [Polls]
Looking for feedback, please make your voice heard!

**Comment by aiceball** [comment]
Hi,

I've been trying out lowkey-vault to test an AKV-bound service, and it's about 90% fitting to the use case, very convenient, so first of all thank you for your work!

One issue I do have, however, is that to my understanding the vault is defined ONLY through the keyvault URI used. If I have a container on the same docker network communicating with lowkey-vault, the vault I'll be hitting will be based on the internal network name I gave that container, i.e. `akv`. If I try to write content to the lowkey-vault from a local python-client, however, the address is different, because I am communicating with lowkey-vault from outside the docker network:
```mermaid
graph
    Python-client -- https://localhost:8443 --> Docker-Lowkey-Vault
    Docker-Service -- https://akv:8443 --> Docker-Lowkey-Vault
```

I wonder if some sort of vault aliasing feature can be added on the management API layer, either at creation time:
```bash
POST /management/vault

{
  "baseUri": "https://bbv:8443",
  "aliasUri": "https://localhost:8443",
  "recoveryLevel": "CustomizedRecoverable+Purgeable",
  "recoverableDays": 42,
  "created": 1641092645,
  "deleted": 1641092645
}
```

or an additional modification point:

```bash
PUT /management/vault/{baseUri}

{
 "aliasUri": "https://{newUri}:8443"
}
```

What this would enable is an access to the same keyvault from multiple networks, i.e. a local network, an internal docker network, an ngrok tunnel, etc.

I am using the Azure Python library to communicate with the Lowkey-Vault if that information is relevant.

**Reply by nagyesta** [reply]
Hi @aiceball ,
first of all, thank you for taking the time and filling out this survey! Also, thank you for providing such detailed instructions about your improvement idea! I love it, will create an issue based on this and figure out how to make it work!

**Reply by nagyesta** [reply]
Created #298 to cover this.

**Comment by conblem** [comment]
We are using lowkey-vault for local testing of our .netcore api's. Some of our devs use M1 MacBooks and docker has to resort back to amd64 emulation on these devices, because there are no native images. The startup time of our whole stack depends on first having a running lowkey-vault instance. At the moment this is really slow for our engineers using ARM machines. We would really appreciate a ARM build of your project.

**Reply by nagyesta** [reply]
Hi @conblem ,
thank you for using Lowkey Vault as well as taking the time to provide valuable feedback! I have to read up on the topic, because I have only known that M1 Macs exist, but never used them (or written any code specifically for them). I assume this should be a problem other projects had to figure out as well, so I am positive there must be a solution. As I am not able to test the thing, would you be willing to provide feedback on the work in progress solution if I manage to figure out something that is supposed to work?
Anyway, I will create an issue in the backlog to be able to track this and will give it a try when I can. In the meantime, if pure Java is an option for you, you could maybe try running the app jar and see whether that is any faster. Hopefully the JDK is already supporting M1.
Thanks again!

**Reply by nagyesta** [reply]
You can track #343 for more details.

**Comment by abtris** [comment]
Hi, great tool. I tried use by https://github.com/nagyesta/lowkey-vault-example-go/ and find that example isn't compatible with latest version (https://github.com/nagyesta/lowkey-vault-example-go/issues/100). Can I just upgrade docker images to your latest version to make it working?

**Reply by nagyesta** [reply]
Hi @abtris ,
thank you for your feedback! Created a new ticket to track this issue. Please find it here: https://github.com/nagyesta/lowkey-vault/issues/697

**Comment by JoeS-Bosch** [comment]
Hi @nagyesta ,
your lowkeyvault is awesome and it helps us a lot in our tests! The LowKeyVault is started as a docker container.

Is there a way to purge the LowKeyVault completely? Right now, I am restarting the container completely, so every secret I created is gone. This is quite time consuming and on the pipeline, SSL errors occurr. I was unable to narrow it down, it might be that I need to add a sleep after the restart delaying it more.

**Reply by nagyesta** [reply]
Hi @JoeS-Bosch ,

thank you very much for your feedback! I think the Management endpoints might be exactly what you need: https://github.com/nagyesta/lowkey-vault/tree/main?tab=readme-ov-file#management-api
There should be one which is deleting a vault and another that can purge it. This way the container can remain alive, but the contents should be gone. 

Please feel free to open an issue in case it is not working as expected, happy to help!

**Reply by JoeS-Bosch** [reply]
It works with Delete, Purge, then Create. Thank you very much!

One minor issue:
All your endpoint require Content-Type to be set to 'application/json'. While swagger does this implicitely, you have to create a workaround for programmatic HttpClients (e.g. in C#, I have to add an empty StringContent to the body).
Normally, get and delete requests, don't need a body or this type of header.

**Reply by nagyesta** [reply]
Hi @JoeS-Bosch ,
thank you for letting me know! That is a bug in my opinion, so I have created #1040 to track and fix it. Please feel free to follow the updates on that ticket.
